### DISCUSSION TITLE
Welcome to smooth-infinite-zoom Discussions!

### DISCUSSION THREAD

**Discussion by BalintKomjati** [BalintKomjati/smooth-infinite-zoom#1] [Announcements]
Questions, feature requests, ideas ... throw them here

**Comment by XeonG** [comment]
anyway to run this locally or something similar that will run locally?

**Reply by BalintKomjati** [reply]
i do not have GPU on my workstation so could not test it, but i do not see any reason why it would not run
have you tried it?
you need a decent GPU with 10+GB vram for this though

**Reply by XeonG** [reply]
oh.. what are the setup instructions to doing locally on windows?  never tried collab thing

**Reply by BalintKomjati** [reply]
It is just a jupyter notebook, you need e.g. VSCode installed with python extensions. 
But based on your questions I feel like running it on colab will be much easier for you.
Go for it! That is well tested.

**Reply by XeonG** [reply]
ah cool I got it working locally..

pretty cool.. I guess will there be any further enhancements to it? not much experience in python.. but things like being able to supply images instead?

**Reply by BalintKomjati** [reply]
great!
if you post something nice please let me know, i am very interested what people craft with this.

i am not actively working on any enhancement atm but will probably get back to it later on...

with regards to using your own image someone already asked that on an other forum, here is my reply:

" you can tweak this with marginal python knowledge by simply replacing init_images manually before the "generate video" block. But i would expect the result will not be very coherent since in the original setup both the init image and the outpainting is generated using the same prompt. anyhow i would love to hear back if you managed to try this out!"

**Reply by XeonG** [reply]
yeah little python knowledge and seems like I don't get any intellisense etc from the language to provide some assistance with ipynb files.. like other language projects... as a feature I would like it if the initial images could be saved with filename bitlike automatic1111 webui along with embedded png meta info of the positive/negative prompts and other values like seed info ... having those orginal images saved is always useful maybe saved file name based on the part of positive prompt.. and having the embedded info is better.. currently can save the 2joined images as a output.png but its really terrible and done via code as an option would be nice...

I suppose also just more options in the script where if resolution is set to 768 or 1024 the mask width is adjusted accordingly.. not sure on best values I had started using 768 for width and height and mask width of 224? would that be right?  I still notice some slight square image borders so maybe its not ideal

I did do this change.. sometimes zooming out worked better, options would be having the start and end frames duplicated for x amount so its pauses at start like 10frames and maybe pauses at end for 30frames.. options like that added in would be nice probably something I'd add 

video_file_name = "infinite_zoom" #@param
fps = 30 #@param
now = datetime.now()
date_time = now.strftime("%m-%d-%Y_%H-%M-%S")
write_video(video_file_name + "_out_"+date_time+".mp4", all_frames, fps, False)
write_video(video_file_name + "_in_"+date_time+".mp4", all_frames, fps, True)

am planning on posting some of the infinite zoom vids on my instagram will let you know it when I do

**Reply by BalintKomjati** [reply]
saving the init images plus the videos like you recommend makes a lot of sense.
could you please create a pull request?

adjusting mask width along with resolution is also great idea. however the notebook imo currently works only with 512 resolution, because any "non-native" resolution destroys its imagination significantly as per my experience. Did you manage to craft something nice outside 512 res? 

My idea to overcome this issue is to generate the video at 512 but upscale the images with SD. There is a branch dedicated to this in the repo. It is an early prototype, it works, but needs further development because "interpolated" images are blurred...

**Comment by XeonG** [comment]
A pull request?

also the png 'parameters' meta text like this would be very helpful if saved from this just for using in Automatic1111 webui...
![tweakpng_y7F2lKzeNJ](https://user-images.githubusercontent.com/1556183/223132801-f1d2b9ca-0d96-45c1-9b1f-b94d37118c46.png)

**Reply by BalintKomjati** [reply]
Are you asking what a pull request is? 
My assumption is that you are not familiar with Github so much so here is a short intro:

In this open source collaborative development setting we use "git" to track code changes. 
(in brackets you can find the git terminology for certain things from here on)

If you copy (="fork") the source code of this project (="repo") you can make changes on your own. 
Once you feel your changes are finished you request for the maintainer of the original repo (=me) to adapt (~="pull") your changes. This way you become a contributor to this project. This is called a Pull Request.

So by me asking if you could make a pull request means that i am kindly promoting you the option to adopt these (or some of these) changes by yourself, and once you create a pull request i will review your changes and merge into the project.
This way your name will be always remembered in the open source community ;)
But it is fully OK to say no for this. (however it seems you already did some changes on your own so whatever you developed and make sense i think it is a waste of effort not to merge into the original repo with a Pull Request)

This also implies that if you would like me to adopt these changes it might not happen anytime soon because i am also occupied by an other project.

**Reply by XeonG** [reply]
yeah I know what you meant, just my changes were just on the fly stuff, I've forked it now and added the changes I had made to it.. sorry all in one commit..  hopefully can get saving of some of the png images out with the meta data embedded, maybe also some further changes to it so batches can be done a bit easier and seperated into folders.. and file name based on positive prompts... yeah much like webui, I kinda think this would be a great integrated into Automatics1111 webui really

**Reply by BalintKomjati** [reply]
just merged, thanks!!
h264 does not work on colab so I reverted back to mp4.
(pls do a colab test run next time if you file a PR)
